extends Node

# Oyun baÅŸlangÄ±cÄ±nda cihaz diline gÃ¶re dil ayarla
func _ready():
	set_language_from_system()

func set_language_from_system():
	# CihazÄ±n dilini al (Ã¶rn: "tr_TR", "en_US")
	var system_locale = OS.get_locale()
	var language_code = system_locale.split("_")[0]  # "tr_TR" -> "tr"
	
	print("ğŸŒ Sistem dili: ", system_locale, " -> ", language_code)
	
	# Desteklenen diller: tr, en
	if language_code in ["tr", "en"]:
		TranslationServer.set_locale(language_code)
		print("âœ… Dil ayarlandÄ±: ", language_code)
	else:
		# Desteklenmeyen dil -> Ä°ngilizce varsayÄ±lan
		TranslationServer.set_locale("en")
		print("âš ï¸ Desteklenmeyen dil, Ä°ngilizce kullanÄ±lÄ±yor")

# Manuel dil deÄŸiÅŸtirme (gelecekte ayarlar menÃ¼sÃ¼ iÃ§in)
func set_language(language_code: String):
	if language_code in ["tr", "en"]:
		TranslationServer.set_locale(language_code)
		print("ğŸ”„ Dil deÄŸiÅŸtirildi: ", language_code)
		# TÃ¼m UI'larÄ± gÃ¼ncelle
		get_tree().reload_current_scene()
